### PR TITLE
Data in tables can now be searches for and filtered

### PR DESCRIPTION
**Pull request by mario-elshaer**
User can now search for what they want and the table changes dynamically. There is 1 search field only, as the user will type something there and the table will recognize it automatically.
Let user: Student

### CODE REVIEW

File: client/src/components/DataTable/DataTable.js [hunk 1]
```diff
@@ -31,16 +18,37 @@ const DataTable = (props) => {
...
   );
+  const [searchTerm, setSearchTerm] = useState("");
+
   useEffect(() => {
     const start = currentPage * maxRows;
     const end = currentPage * maxRows + maxRows;
-    setCurrentData(displayedData.slice(start, end) || displayedData);
-  }, [maxRows, currentPage, displayedData]);
+    const filteredData =
+      searchTerm === ""
```
Comment (ramy-att) [inline client/src/components/DataTable/DataTable.js:27]: Thanks for using triple =

File: client/src/components/DataTable/DataTable.js [hunk 2]
```diff
@@ -31,16 +18,37 @@ const DataTable = (props) => {
...
+        ? displayedData
+        : displayedData.filter((row) =>
+            Object.values(row).some((val) =>
+              String(val).toLowerCase().includes(searchTerm.toLowerCase())
```
Comment (ramy-att) [inline client/src/components/DataTable/DataTable.js:31]: This code is so clean. How did you come up with it?

**Review by ramy-att — APPROVED** [review]
Tamam

File: client/src/components/DataTable/DataTable.js (hunk 2 above)
Comment (mario-elshaer) [reply client/src/components/DataTable/DataTable.js:31]: Persistance and Perseverance

**Pull request merged**